src/main/python/pgshovel/replication/streams/kafka.py

```diff
@@ -40,12 +54,44 @@ def consume(self, state):
             consumer.seek(delta, 1)
             assert consumer.offsets[0] == offset
 
-        for offset, message in consumer:
-            decoded = self.codec.decode(message.value)
-            state = validate_state(state, offset, decoded)
+        for offset, message in self.__primed(consumer, state):
```

Comment:
Won't this trigger priming each time the process starts (rather than only if the consumer has no existing stream state?)

This scenario seems problematic: a consumer encounters an invalid transition, e.g. `Commit` to `Rollback` or `Commit` to `Mutation` and crashes with an `InvalidEventError`. (As an aside: this should be replaced with `InvalidOperationError` or `InvalidMessageError` accounting for prior conversations.) Then, the consumer restarts and primes the stream at startup from the last commit point, discarding invalid events that were legitimately invalid and artifacts of stream corruption, potentially losing data.

An example of this would be if the `Begin` message wasn't published for some reason (that possibly gets caught in sequence validation, but I wouldn't want that to be the sole defense here) and that _entire batch would be skipped_ as the next batch it would still pass validation: https://github.com/disqus/pgshovel/blob/replication/src/main/python/pgshovel/replication/validation/transactions.py#L100-L105

I _think_ this is only a problem with invalid events after an event that causes a transition to `Committed` or `RolledBack`.